### PR TITLE
Converted donation-modal.scss into Tailwind classes

### PR DESCRIPTION
**Pull request by mmmavis**
Related PRs/issues: #8706 

### To QA
1. Open the [Heroku app link](https://foundation-s-8706-scss--svbqjm.herokuapp.com/en/campaigns/single-page/?thank_you=true) on one browser tab
2. Open this [prod link](https://foundation.mozilla.org/en/campaigns/tell-slack-make-dms-safe/?thank_you=true) on another tab
3. Compare the two donation modals to see if they look the same
4. Using the Heroku sample test page, verify the `x` button and `No, I'll share instead" can both close the modal.

### CODE REVIEW

File: source/js/components/petition/donation-modal.jsx [hunk 1]
```diff
@@ -17,6 +17,17 @@ import classNames from "classnames";
  *   - donateText, the "donate" button label
  *   - shareText, the "no donate, just share" button label
  */
+
+const MODAL_OVERLAY_CLASS = "tw-fixed tw-inset-0 tw-bg-white/90 tw-z-[1050]";
```
Comment (mmmavis) [inline source/js/components/petition/donation-modal.jsx:21]: Having these as variables at the top just for better code readability

File: source/js/components/petition/donation-modal.jsx [hunk 2]
```diff
@@ -58,8 +69,8 @@ class DonationModal extends Component {
   render() {
     return (
       <div ref={(e) => (this.domLocation = e)}>
-        <div className="modal-underlay" ref={(e) => (this.fragment = e)}>
-          <div className="modal show" role="dialog">
+        <div className={MODAL_OVERLAY_CLASS} ref={(e) => (this.fragment = e)}>
+          <div className={classNames(`modal`, MODAL_CLASS)} role="dialog">
```
Comment (mmmavis) [inline source/js/components/petition/donation-modal.jsx:73]: We need to keep `modal` class here as we are still relying on Bootstrap's stylings for `.modal`

File: source/js/components/petition/donation-modal.jsx [hunk 3]
```diff
@@ -74,18 +85,21 @@ class DonationModal extends Component {
...
+          className={CLOSE_BUTTON_CLASS}
           data-dismiss="modal"
           aria-label="Close"
           tabIndex="0"
         >
           <span aria-hidden="true">&times;</span>
         </button>
 
-        <div className="modal-body">
+        <div className={classNames(`modal-body`, MODAL_BODY_CLASS)}>
```
Comment (mmmavis) [inline source/js/components/petition/donation-modal.jsx:102]: Similarly, we need to keep `modal-body` class here as we are still relying on Bootstrap's stylings for `.modal-body`

File: source/js/components/petition/donation-modal.jsx [hunk 4]
```diff
@@ -74,18 +85,21 @@ class DonationModal extends Component {
       this.donateURL = `?form=donate`;
     }
     return (
-      <div className="modal-content" role="dialog">
+      <div
+        className={classNames(`modal-content`, MODAL_CONTENT_CLASS)}
```
Comment (mmmavis) [inline source/js/components/petition/donation-modal.jsx:89]: Similarly, we need to keep `modal-content` class here as we are still relying on Bootstrap's stylings for `.modal-content`

**Review by danielfmiranda — APPROVED** [review]
Looks great! Thanks @mmmavis 🙌 approved

**Pull request merged**